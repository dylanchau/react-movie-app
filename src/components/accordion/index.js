import { createContext, useContext, useState } from 'react'

import {
  Body,
  Container,
  Frame,
  Header,
  Inner,
  Item,
  Title,
} from './styles/accordion'

const ToggleContext = createContext()

export default function Accordion({ children, ...restProps }) {
  return (
    <Container {...restProps}>
      <Inner>{children}</Inner>
    </Container>
  )
}

Accordion.Title = function AccordionTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>
}

Accordion.Frame = function AccordionFrame({ children, ...restProps }) {
  return <Frame {...restProps}>{children}</Frame>
}

Accordion.Item = function AccordionItem({ children, ...restProps }) {
  const [toggle, setToggle] = useState(false)

  return (
    <ToggleContext.Provider value={{ toggle, setToggle }}>
      <Item {...restProps}>{children}</Item>
    </ToggleContext.Provider>
  )
}

Accordion.Header = function AccordionHeader({ children, ...restProps }) {
  const { toggle, setToggle } = useContext(ToggleContext)

  return (
    <Header onClick={() => setToggle((preState) => !preState)} {...restProps}>
      {children}
      {toggle ? (
        <img src="/images/icons/close-slim.png" alt="Close" />
      ) : (
        <img src="/images/icons/add.png" alt="Open" />
      )}
    </Header>
  )
}

Accordion.Body = function AccordionBody({ children, ...restProps }) {
  const { toggle } = useContext(ToggleContext)
  return toggle ? <Body {...restProps}>{children}</Body> : null
}
