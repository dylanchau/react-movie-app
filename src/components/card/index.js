import { createContext, useContext, useState } from 'react'

import {
  Container,
  Content,
  Entities,
  Feature,
  FeatureClose,
  FeatureText,
  FeatureTitle,
  Group,
  Image,
  Item,
  Maturity,
  Meta,
  SubTitle,
  Text,
  Title,
} from './styles/card'

export const FeatureContext = createContext()

export default function Card({ children, ...restProps }) {
  const [showFeature, setShowFeature] = useState(false)
  const [itemFeature, setItemFeature] = useState({})

  return (
    <FeatureContext.Provider
      value={{ showFeature, setShowFeature, itemFeature, setItemFeature }}
    >
      <Container {...restProps}>{children}</Container>
    </FeatureContext.Provider>
  )
}

Card.Group = function CardGroup({ children, ...restProps }) {
  return <Group {...restProps}>{children}</Group>
}

Card.Tittle = function CardTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>
}

Card.SubTittle = function CardSubTittle({ children, ...restProps }) {
  return <SubTitle {...restProps}>{children}</SubTitle>
}

Card.Text = function CardText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>
}

Card.Meta = function CardMeta({ children, ...restProps }) {
  return <Meta {...restProps}>{children}</Meta>
}

Card.Image = function CardImage({ children, ...restProps }) {
  return <Image {...restProps} />
}

Card.Item = function CardItem({ item, children, ...restProps }) {
  const { setItemFeature, setShowFeature } = useContext(FeatureContext)

  return (
    <Item
      onClick={() => {
        setItemFeature(item)
        setShowFeature(true)
      }}
      {...restProps}
    >
      {children}
    </Item>
  )
}

Card.Entities = function CardEntities({ children, ...restProps }) {
  return <Entities {...restProps}>{children}</Entities>
}

Card.Entities = function CardEntities({ children, ...restProps }) {
  return <Entities {...restProps}>{children}</Entities>
}

Card.Feature = function CardFeature({ children, category, ...restProps }) {
  const { showFeature, itemFeature, setShowFeature } = useContext(
    FeatureContext
  )
  return showFeature ? (
    <Feature
      src={`/images/${category}/${itemFeature.genre}/${itemFeature.slug}/large.jpg`}
      {...restProps}
    >
      <Content>
        <FeatureTitle>{itemFeature.title}</FeatureTitle>
        <FeatureText>{itemFeature.description}</FeatureText>
        <FeatureClose onClick={() => setShowFeature((prev) => !prev)}>
          <Image src="/images/icons/close.png" alt="Close" />
        </FeatureClose>

        <Group margin="30px 0" flexDirection="row" alginItem="center">
          <Maturity rating={itemFeature.maturity}>
            {itemFeature.maturity < 12 ? 'PG' : itemFeature.maturity}
          </Maturity>
          <FeatureText fontWeight="bold">
            {itemFeature.genre.charAt(0).toUpperCase() +
              itemFeature.genre.slice(1)}
          </FeatureText>
        </Group>
        {children}
      </Content>
    </Feature>
  ) : null
}
