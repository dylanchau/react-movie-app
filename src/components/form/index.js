import {
  Base,
  Container,
  Error,
  Input,
  Link,
  Submit,
  Text,
  TextSmall,
  Title,
} from './styles/form'

export default function Form({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>
}

Form.Title = ({ children, ...restProps }) => (
  <Title {...restProps}>{children}</Title>
)

Form.Base = ({ children, ...restProps }) => (
  <Base {...restProps}>{children}</Base>
)

Form.Text = ({ children, ...restProps }) => (
  <Text {...restProps}>{children}</Text>
)

Form.TextSmall = ({ children, ...restProps }) => (
  <TextSmall {...restProps}>{children}</TextSmall>
)

Form.Submit = ({ children, ...restProps }) => (
  <Submit {...restProps}>{children}</Submit>
)

Form.Error = ({ children, ...restProps }) => (
  <Error {...restProps}>{children}</Error>
)

Form.Link = ({ children, ...restProps }) => (
  <Link {...restProps}>{children}</Link>
)

Form.Input = ({ children, ...restProps }) => (
  <Input {...restProps}>{children}</Input>
)
