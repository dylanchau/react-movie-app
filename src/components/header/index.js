import { useState } from 'react'
import { Link as ReactRouterLink } from 'react-router-dom'

import {
  Background,
  ButtonLink,
  Container,
  Dropdown,
  Feature,
  FeatureCallOut,
  Group,
  Link,
  Logo,
  Picture,
  PlayButton,
  Profile,
  Search,
  SearchIcon,
  SearchInput,
  Text,
} from './styles/header'

export default function Header({ bg = true, children, ...restProps }) {
  return bg ? <Background {...restProps}>{children}</Background> : children
}

Header.Frame = function HeaderFrame({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>
}

Header.Group = function HeaderGroup({ children, ...restProps }) {
  return <Group {...restProps}>{children}</Group>
}

Header.Search = function HeaderSearch({
  searchTerm,
  setSearchTerm,
  ...restProps
}) {
  const [searchActive, setSearchActive] = useState(false)

  return (
    <Search {...restProps}>
      <SearchIcon
        onClick={() => setSearchActive((preProp) => !preProp)}
        data-testid="search-click"
      >
        <img alt="search" src="/images/icons/search.png" />
      </SearchIcon>
      <SearchInput
        value={searchTerm}
        onChange={({ target }) => setSearchTerm(target.value)}
        active={searchActive}
        placeholder="Search..."
        data-testid="search-input"
      />
    </Search>
  )
}

Header.ButtonLink = function HeaderButtonLink({ children, ...restProps }) {
  return <ButtonLink {...restProps}>{children}</ButtonLink>
}

Header.Logo = function HeaderLogo({ to, ...restProps }) {
  return (
    <ReactRouterLink to={to}>
      <Logo {...restProps} />
    </ReactRouterLink>
  )
}

Header.Feature = function HeaderFeature({ children, ...restProps }) {
  return <Feature {...restProps}>{children}</Feature>
}

Header.FeatureCallOut = function HeaderFeatureCallOut({
  children,
  ...restProps
}) {
  return <FeatureCallOut {...restProps}>{children}</FeatureCallOut>
}

Header.Text = function HeaderText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>
}

Header.TextLink = function LinkText({ children, ...restProps }) {
  return <Link {...restProps}>{children}</Link>
}

Header.Profile = function HeaderProfile({ children, ...restProps }) {
  return <Profile {...restProps}>{children}</Profile>
}

Header.Dropdown = function HeaderDropdown({ children, ...restProps }) {
  return <Dropdown {...restProps}>{children}</Dropdown>
}

Header.Picture = function HeaderPicture({ src, ...restProps }) {
  return <Picture {...restProps} src={`/images/users/${src}.png`} />
}

Header.PlayButton = function HeaderPlayButton({ children, ...restProps }) {
  return <PlayButton {...restProps}>{children}</PlayButton>
}
