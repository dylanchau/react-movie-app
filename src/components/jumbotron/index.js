import { Inner } from './styles/jumbotron';

export default function Jumbotron() {
  return (
    <Inner>
      <p>Hello</p>
    </Inner>
  );
}
