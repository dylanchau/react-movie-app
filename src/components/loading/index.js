import { LockBody, Picture, ReleaseBody, Spinner } from './styles/loading'

function Loading({ src, ...restProps }) {
  return (
    <Spinner {...restProps}>
      <LockBody />
      <Picture src={`/images/users/${src}.png`} alt={src} />
    </Spinner>
  )
}

Loading.ReleaseBody = function LoadingReleaseBody({ ...restProps }) {
  return <ReleaseBody {...restProps} />
}

export default Loading
