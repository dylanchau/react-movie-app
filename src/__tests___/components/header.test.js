import { fireEvent, render } from '@testing-library/react'
import { Header } from 'components'
import logo from 'logo.svg'
import { MemoryRouter } from 'react-router-dom'

describe('<Header /> component', () => {
  it('Render header component - only include Logo and Button', () => {
    const { container, getByText, getByAltText } = render(
      <Header>
        <Header.Frame>
          <Header.Logo to="#" alt="Netflix" src={logo} />
          <Header.ButtonLink to="#">Sign In</Header.ButtonLink>
        </Header.Frame>
      </Header>,
      { wrapper: MemoryRouter }
    )

    expect(getByAltText('Netflix')).toBeTruthy()
    expect(getByText('Sign In')).toBeTruthy()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('Render header component full populate', () => {
    const { container, getByText, getByAltText, getByTestId } = render(
      <Header src="joker1" notShowOnSmallViewPort>
        <Header.Frame>
          <Header.Group>
            <Header.Logo to="#" src={logo} alt="film" />
            <Header.TextLink onClick={() => {}} active>
              Series
            </Header.TextLink>
            <Header.TextLink onClick={() => {}}>Films</Header.TextLink>
          </Header.Group>

          <Header.Group>
            <Header.Search searchTerm="Joker" setSearchTerm={() => {}} />
            <Header.Profile>
              <Header.Picture src="" />
              <Header.Dropdown>
                <Header.Group>
                  <Header.Picture src="" />
                  <Header.TextLink>Tester</Header.TextLink>
                </Header.Group>

                <Header.Group>
                  <Header.TextLink onClick={() => {}}>Sign Out</Header.TextLink>
                </Header.Group>
              </Header.Dropdown>
            </Header.Profile>
          </Header.Group>
        </Header.Frame>

        <Header.Feature>
          <Header.FeatureCallOut>Watch Joker Now</Header.FeatureCallOut>
          <Header.Text>Header feature text</Header.Text>
          <Header.PlayButton>Play</Header.PlayButton>
        </Header.Feature>
      </Header>,
      { wrapper: MemoryRouter }
    )

    expect(getByTestId('search-input')).toBeTruthy()
    expect(getByTestId('search-input').value).toBe('Joker')
    fireEvent.change(getByTestId('search-input'), {
      target: { value: 'Simpsons' },
    })
    fireEvent.click(getByTestId('search-click'))

    expect(getByAltText('film')).toBeTruthy()
    expect(getByText('Series')).toBeTruthy()
    expect(getByText('Films')).toBeTruthy()
    expect(getByText('Tester')).toBeTruthy()
    expect(getByText('Sign Out')).toBeTruthy()
    expect(getByText('Watch Joker Now')).toBeTruthy()
    expect(getByText('Header feature text')).toBeTruthy()
    expect(getByText('Play')).toBeTruthy()

    expect(container.firstChild).toMatchSnapshot()
  })
})
