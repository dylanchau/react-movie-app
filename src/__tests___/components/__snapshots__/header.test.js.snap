// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> component Render header component - only include Logo and Button 1`] = `
<div
  class="header__Background-q26fio-0 dbrQnf"
>
  <div
    class="header__Container-q26fio-1 boVZsC"
  >
    <a
      href="/"
    >
      <img
        alt="Netflix"
        class="header__Logo-q26fio-14 jonBRx"
        src="logo.svg"
      />
    </a>
    <a
      class="header__ButtonLink-q26fio-7 kARkTh"
      href="/"
    >
      Sign In
    </a>
  </div>
</div>
`;

exports[`<Header /> component Render header component full populate 1`] = `
<div
  class="header__Background-q26fio-0 kgLMhA"
  src="joker1"
>
  <div
    class="header__Container-q26fio-1 boVZsC"
  >
    <div
      class="header__Group-q26fio-3 jseyGL"
    >
      <a
        href="/"
      >
        <img
          alt="film"
          class="header__Logo-q26fio-14 jonBRx"
          src="logo.svg"
        />
      </a>
      <p
        class="header__Link-q26fio-2 jXeTQa"
      >
        Series
      </p>
      <p
        class="header__Link-q26fio-2 jXeTQa"
      >
        Films
      </p>
    </div>
    <div
      class="header__Group-q26fio-3 jseyGL"
    >
      <div
        class="header__Search-q26fio-5 bdgbQD"
      >
        <button
          class="header__SearchIcon-q26fio-6 buwzOa"
          data-testid="search-click"
        >
          <img
            alt="search"
            src="/images/icons/search.png"
          />
        </button>
        <input
          class="header__SearchInput-q26fio-4 enOho"
          data-testid="search-input"
          placeholder="Search..."
          value="Joker"
        />
      </div>
      <div
        class="header__Profile-q26fio-10 gCSUAh"
      >
        <button
          class="header__Picture-q26fio-8 hAZzGM"
          src="/images/users/.png"
        />
        <div
          class="header__Dropdown-q26fio-9 bVGpQy"
        >
          <div
            class="header__Group-q26fio-3 jseyGL"
          >
            <button
              class="header__Picture-q26fio-8 hAZzGM"
              src="/images/users/.png"
            />
            <p
              class="header__Link-q26fio-2 jXeTQa"
            >
              Tester
            </p>
          </div>
          <div
            class="header__Group-q26fio-3 jseyGL"
          >
            <p
              class="header__Link-q26fio-2 jXeTQa"
            >
              Sign Out
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="header__Container-q26fio-1 header__Feature-q26fio-11 boVZsC coJtYq"
  >
    <h2
      class="header__FeatureCallOut-q26fio-12 jHPtTM"
    >
      Watch Joker Now
    </h2>
    <p
      class="header__Text-q26fio-13 jtQKvQ"
    >
      Header feature text
    </p>
    <button
      class="header__PlayButton-q26fio-15 eNMWyZ"
    >
      Play
    </button>
  </div>
</div>
`;
